dialogflow/stopwatch/README.md

```diff
@@ -1,42 +1,72 @@
 
-## Prepare your Google Cloud account
-As with all Google Cloud APIs, every call to the Dialogflow API must be associated
-  with a project within the [Google Cloud Console][cloud-console] that has the
-  API enabled. In brief:
-  - Create a project (or use an existing one) in the [Cloud Console][cloud-console]
-  - [Enable billing][billing].
-
-## Enable Dialogflow
-If you have not already done so, [enable Dialogflow for your project](https://cloud.google.com/dialogflow-enterprise/docs/quickstart). Scripts related to the quickstart are in the SETUP-SERVICE directory.
-
-## Running the app
-- In the [Google Cloud Console](https://console.cloud.google.com), use the APIs&Services->Library menu item to find and enable the Dialogflow API.
-- Also using the Credentials item in the Google Cloud Console, create a Service Account key. Use the popup to create a key for a "New service account", name it "stopwatch" (this is arbitrary), and give it the role of `Dialogflow API Admin`. Choose a key type of "JSON" and press the "Create" button to download a key file. As a convenience, rename this file to `credentials.json`.
-- Clone this repository and `cd` into this directory.
-- Copy your `credentials.json` over the skeletal version in this directory. It is used by the scripts in `SETUP_SERVICE` and by the sample application. 
-- Be sure that you have gone through the steps in the [Dialogflow quickstart](https://cloud.google.com/dialogflow-enterprise/docs/quickstart) to create and configure your stopwatch agent. Helper scripts for this are in the `SETUP_SERVICE` directory.
-- Run ./INSTALL-COCOAPODS to install app dependencies. When it finishes, it will open the Stopwatch workspace in Xcode. Since we are using Cocoapods, be sure to open the workspace and not Stopwatch.xcodeproj.
-- Replace `your-project-identifier` in `StopwatchService.swift` with the identifier of your Google Cloud project.
+## Setup
+- Create a project with the [Google Cloud Console][cloud-console]
+- Enable the [Dialogflow API](https://console.cloud.google.com/apis/library/dialogflow.googleapis.com).
+- [Enable billing][billing].
+- [Import the Dialogflow Agent](https://dialogflow.com/docs/agents/export-import-restore#import) using the `StopwatchAgent.zip` which is located in the `stopwatch` directory. 
+- [Create a Service account](https://cloud.google.com/iam/docs/creating-managing-service-accounts) with the following IAM role: `Dialogflow API Client`. Example name: `dialogflow-client`. ([For more info on: how to add roles to a Service Account](https://cloud.google.com/iam/docs/granting-roles-to-service-accounts#granting_access_to_a_service_account_for_a_resource))
+- For your "App Engine Default Service Account" add the following role `Service Account Token Creator` Role. ([For more info on: how to add roles to a Service Account](https://cloud.google.com/iam/docs/granting-roles-to-service-accounts#granting_access_to_a_service_account_for_a_resource))
```

Comment:
Firebase functions side we need two different service accounts one to create credentials and one to create proxy credentials... and it seems firebase functions runs using the google application default service accounts and that's the one which we are using to generate the access token. you can find this in the IAM section similar to the attached image.
<img width="1385" alt="App Engine default service account" src="https://user-images.githubusercontent.com/28930192/55281345-3cf79d00-52f0-11e9-9074-181b465df392.png">

Reply:
To find the "App Engine Default Service Account", Please finish the "Setup and Deploy the Firebase Function" section in README.md line 46.  To avoid this confusion i have updated the README.md with the appropriate sequence.